### DISCUSSION TITLE
Server-side diff for "argocd app diff --local"

### DISCUSSION THREAD

**Discussion by MPV** [argoproj/argo-cd#7524] [Q&A]
Does the [`argocd app diff --local` command](https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_app_diff/) support doing the equivalent of a `kubectl diff --server-side=true`?

Would be nice being able to check that a change is valid according to installed CRDs before trying to deploy invalid changes.

**Comment by pacoguzman** [comment]
Hi there, I try to do that exact same thing. Validating pushed applications to the config repository in topic branches. Once merge we'll made the sync

**Comment by rishabh625** [comment]
Hi,
``` argocd app diff --local``` just compares current live state to decalarative config present locally instead of git repository.
So yes you can view and validate diff,  that will be applied to kubernetes cluster same as kubectl diff before commiting to git repo and syncing.

eg
``` argocd app diff testmanifest  --local /home/risha/dev --local-repo-root ~/ ```

Unlike  --server-side=true validations are not provided

**Comment by pacoguzman** [comment]
Thank you understood.
